OutOfSchool/OutOfSchool.AuthServer.Tests/EmailSender/SendGridAccessibilityServiceTests.cs

```diff
@@ -83,4 +83,32 @@ public void IsSendGridAccessible_WithOptionsTimeoutTimeNull_ReturnsTrueAfterTime
         Assert.True(earlyResult);
         Assert.True(result);
     }
+
+    [Test]
+    public void GetAccessibilityTime_WithAccessibleSendGrid_ReturnsDateTimeOffsetNow()
+    {
+        // Arrange
+        var now = DateTimeOffset.Now;
+
+        // Act
+        var accessTime = _emailService.GetAccessibilityTime(now);
+
+        // Assert
+        Assert.AreEqual(now, accessTime);
+    }
+
+    [Test]
+    public void GetAccessibilityTime_WithInaccessibleSendGrid_ReturnsAccessibilityTime()
+    {
+        // Arrange
+        var now = DateTimeOffset.Now;
+
+        // Act
+        _emailService.SetSendGridInaccessible(now);
+        var accessTime = _emailService.GetAccessibilityTime(now);
+
+        // Assert
+        Assert.AreNotEqual(now, accessTime);
+        Assert.AreEqual(now.AddMinutes(300), accessTime);
```

Comment:
(Fragile test) This test is checking that next access time should be higher than now. We don't know in test the precise difference (currently used 5hrs could be changed in implementation), and shouldn't know/rely on it. 

That's why please assert only one condition `now < accessTime`.